### PR TITLE
Fixed #28680 -- Doc'd Func.__init__()'s **extra and as_sql()'s **extra_context aren't escaped.

### PR DESCRIPTION
**Pull request by timgraham**
https://code.djangoproject.com/ticket/28680

### CODE REVIEW

File: docs/ref/models/expressions.txt [hunk 1]
```diff
@@ -289,7 +289,11 @@ The ``Func`` API is as follows:
 
         The ``as_vendor()`` methods should use the ``function``, ``template``,
         ``arg_joiner``, and any other ``**extra_context`` parameters to
-        customize the SQL as needed. For example:
+        customize the SQL as needed. ``extra_context`` must not contain
+        untrusted user input as these values aren't escaped when they're
```
Comment (shaib) [inline docs/ref/models/expressions.txt]: I sort of agree with @hynekcer . I'd replace "these values aren't escaped" with something like "these values are interpolated into the SQL string, not passed as parameters, and aren't even escaped".

**Comment by timgraham** [comment]
Hynek, I've tried to address the suggestions you left on the ticket.

**Comment by hynekcer** [comment]
Tim, thanks for the fix.

I see still what @shaib objected above. Maybe it is a detail. Three times an obsoleted incorrect text "passed as query parameters, where the database would escape them." This worst solution would be if only the framework implements parameterized queries, but an SQL driver didn't support them yet. I also wrote about escaping incorrectly at the beginning. The parameterized query should be explained at least once briefly and correctly. (Otherwise it looks like Django backends use obsoleted practices, but it is OK.)

I like "**protected by strict separation SQL code and data**", that I concentrated from this text:
> Query parameterization refers to the process of building database queries in application code in a specialized way. Query parameterization first defines all static SQL code, and then passes in each parameter to the query in a separate section of code. This coding style allows the database to distinguish between code and data, regardless of what user input is supplied, and successfully defends against SQL Injection. ([Query Parameterization - OWASP](https://www.owasp.org/index.php/Query_Parameterization))

I found a short clear text about query parameterization at [microsoft blog](https://blogs.msdn.microsoft.com/sqlphp/2008/09/30/how-and-why-to-use-parameterized-queries/) "... a query plan is constructed on the server before the query is executed with parameter values".

**Comment by timgraham** [comment]
If the text reads, "where the database driver would escape them" (addition of 'driver'), does that address your concern?

**Comment by hynekcer** [comment]
I aggree. You are right, exactly with this issue. I found that my previous post was not relevant to Django database drivers. I checked up it yesterday for psycopg2 by reading many of its C source. Please, let me take a short time to think. Escaping of parameters is a safe technique if it is combined with a carefully constructed SQL and processed by a database driver. That is still OK for ORM, but generally not for manually constructed SQL. Maybe I contact you again by a security group email. This issue can be closed by your commits above as you think it as appropriate.

**Pull request merged**